ts/Stock/Indicators/Klinger/KlingerIndicator.ts

```diff
@@ -0,0 +1,434 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type {
+    KlingerOptions,
+    KlingerParamsOptions
+} from './KlingerOptions';
+import type KlingerPoint from './KlingerPoint';
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import RequiredIndicatorMixin from '../../../Mixins/IndicatorRequired.js';
+import MultipleLinesMixin from '../../../Mixins/MultipleLines.js';
+import SeriesRegistry from '../../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator,
+        ema: EMAIndicator
+    }
+} = SeriesRegistry;
+import U from '../../../Core/Utilities.js';
+const {
+    correctFloat,
+    extend,
+    isArray,
+    merge
+} = U;
+
+/**
+ * The Klinger Indicator series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.klinger
+ *
+ * @augments Highcharts.Series
+ */
+class KlingerIndicator extends SMAIndicator {
+    /**
+     * Klinger indicator. This series requires the `linkedTo` option to be set
+     * and should be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/klinger
+     *         Klinger indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/klinger
+     * @optionparent plotOptions.klinger
+     */
+    public static defaultOptions: KlingerOptions = merge(SMAIndicator.defaultOptions, {
+        /**
+         * Paramters used in calculation of Klinger Oscillator.
+         *
+         * @excluding index, period
+         */
+        params: {
+            /**
+             * The fast period for indicator calculations.
+             */
+            fastAvgPeriod: 34,
+            /**
+             * The slow period for indicator calculations.
+             */
+            slowAvgPeriod: 55,
+            /**
+             * The base period for signal calculations.
+             */
+            signalPeriod: 13,
+            /**
+             * The id of another series to use its data as volume data for the
+             * indiator calculation.
+             */
+            volumeSeriesID: 'volume'
+        },
+        signal: {
+            /**
+             * Styles for a signal line.
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * [plotOptions.klinger.color
+                 * ](#plotOptions.sma.color).
```

Comment:
`#plotOptions.sma.color` or `#plotOptions.klinger.color`?

Reply:
Is signalLine's color inherited from SMA? I believe it should be inherited from Klinger indicator?